corehq/ex-submodules/couchforms/tests/test_geopoint.py

```diff
@@ -1,48 +1,42 @@
 from decimal import Decimal
 
+import pytest
 from jsonobject.exceptions import BadValueError
 from nose.tools import assert_equal, assert_raises
 
 from ..geopoint import GeoPoint
 
 
-def test_valid_geopoint_properties():
-    for input_string, output in [
-        ('42.3739063 -71.1109113 0.0 886.0', ('42.3739063', '-71.1109113', '0.0', '886.0')),
-        ('-7.130 -41.563 7.53E-4 8.0', ('-7.130', '-41.563', '0', '8.0')),
-        ('-7.130 -41.563 -2.2709742188453674E-4 8.0', ('-7.130', '-41.563', '0', '8.0')),
-        ('-7.130 -41.563', ('-7.130', '-41.563', 'NaN', 'NaN')),
-        ('-7.130 -41.563 1.2E-3 0', ('-7.130', '-41.563', '0.012', '0')),
-        ('-7.130 -41.563 0.0 1.0', ('-7.130', '-41.563', '0.0', '1.0')),
-    ]:
-        actual = GeoPoint.from_string(input_string, flexible=True)
-        expected = GeoPoint(*(Decimal(x) for x in output))
-    yield _geopoints_equal, actual, expected
+@pytest.mark.parametrize("input_string, output", [
+    ('42.3739063 -71.1109113 0.0 886.0', ('42.3739063', '-71.1109113', '0.0', '886.0')),
+    ('-7.130 -41.563 7.53E-4 8.0', ('-7.130', '-41.563', '0', '8.0')),
+    ('-7.130 -41.563 -2.2709742188453674E-4 8.0', ('-7.130', '-41.563', '0', '8.0')),
+    ('-7.130 -41.563', ('-7.130', '-41.563', 'NaN', 'NaN')),
+    ('-7.130 -41.563 1.2E-3 0', ('-7.130', '-41.563', '0.0012', '0')),
```

Comment:
Do you remember why the `0.012` changed?

Reply:
Previously the indentation had a bug: the `yield` was outside the `for` loop. So the generator was only yielding a single test, the last item in the list of cases. In other words, this particular case was not being tested.

I'm fairly confident this change is correct since `1.2E-3 == 0.0012` (not `0.012`).

Reply:
thanks, that makes sense